### ISSUE TITLE
[TSX] Arrow function's return type annotation breaks the Syntax Highlighting.

### ISSUE THREAD

**Issue by ivansvlv** [sublimehq/Packages#2573]
- Sublime Version: Build 4090
- OS Version: macOS Catalina 10.15.7

Without type annotation:

![image](https://user-images.githubusercontent.com/16175855/98288530-a8ffb400-1faf-11eb-9af7-f5492f05d389.png)

With type annotation (already broken, note the colour of identifier, not green anymore):

![image](https://user-images.githubusercontent.com/16175855/98288637-d0ef1780-1faf-11eb-9a51-5f989baa26bc.png)

With type `void` (this will actually break syntax significantly in a function's body and further in the code, looks like `void` is recognized as an operator here):

![image](https://user-images.githubusercontent.com/16175855/98288957-3216eb00-1fb0-11eb-9be3-b84e8a8e6523.png)

Please let me know if more information is needed. Thank you.

**Comment by rchl** [comment]
Looks very much like https://github.com/sublimehq/Packages/issues/2468 but it's possible that your issue touches on something extra (at least the `void` thing).

**Comment by ivansvlv** [comment]
@rchl yes indeed! looks similar. damn, i did few quick searches before reporting it, should have been more diligent.

**Comment by Thom1729** [comment]
I think this is definitely a duplicate of #2468 and should probably be closed.

On the bright side, https://github.com/sublimehq/Packages/pull/2574 should hopefully resolve this.

**Comment by ivansvlv** [comment]
Guys, should I close this issue, or it will be automatically closed once PR is merged?

**Comment by Thom1729** [comment]
Since there's another issue already open, this one should probably be closed.

**Comment by ivansvlv** [comment]
ℹ️ Issue closed as duplicate of https://github.com/sublimehq/Packages/issues/2468
